File: wrappers/dotnet/indy-sdk-dotnet-test/NonSecretsTests/AddRecordTest.cs
using Hyperledger.Indy.NonSecretsApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.NonSecretsTests
{
    [TestClass]
    public class AddRecordTest : NonSecretsIntegrationTestBase
    {
        [TestMethod]
        public async Task TestAddRecordWorks()
        {
            await NonSecrets.AddRecordAsync(wallet, type, id, value, tagsEmpty);
        }

        [TestMethod]
        public async Task TestAddRecordWorksForDifferentIds()
        {
            await NonSecrets.AddRecordAsync(wallet, type, id, value, tagsEmpty);
            await NonSecrets.AddRecordAsync(wallet, type, id2, value, tagsEmpty);
        }

        [TestMethod]
        public async Task TestAddRecordWorksForDuplicate()
        {
            await NonSecrets.AddRecordAsync(wallet, type, id, value, tagsEmpty);

            var ex = await Assert.ThrowsExceptionAsync<WalletItemAlreadyExistsException>(() =>
                NonSecrets.AddRecordAsync(wallet, type, id, value, tagsEmpty)
            );

        }
    }
}

File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/non_secrets/AddRecordTest.java
package org.hyperledger.indy.sdk.non_secrets;

import org.hyperledger.indy.sdk.wallet.WalletItemAlreadyExistsException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;


public class AddRecordTest extends NonSecretsIntegrationTest {

	@Test
	public void testAddRecordWorks() throws Exception {
		WalletRecord.add(wallet, type, id, value, tagsEmpty).get();
	}

	@Test
	public void testAddRecordWorksForDifferentIds() throws Exception {
		WalletRecord.add(wallet, type, id, value, tagsEmpty).get();
		WalletRecord.add(wallet, type, id2, value, tagsEmpty).get();
	}

	@Test
	public void testAddRecordWorksForDuplicate() throws Exception {
		WalletRecord.add(wallet, type, id, value, tagsEmpty).get();

		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletItemAlreadyExistsException.class));

		WalletRecord.add(wallet, type, id, value, tagsEmpty).get();
	}
}